mrp_lot_number_propagation/models/mrp_bom.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models, tools
+
+
+class MrpBom(models.Model):
+    _inherit = "mrp.bom"
+
+    lot_number_propagation = fields.Boolean(
+        default=False,
+        help=(
+            "Allow to propagate the lot/serial number "
+            "from a component to the finished product."
+        ),
+    )
+    display_lot_number_propagation = fields.Boolean(
+        compute="_compute_display_lot_number_propagation"
+    )
+
+    @api.depends(
+        "type",
+        "product_tmpl_id.tracking",
+        "product_qty",
+        "product_uom_id",
+        "bom_line_ids.product_id.tracking",
+        "bom_line_ids.product_qty",
+        "bom_line_ids.product_uom_id",
+    )
+    def _compute_display_lot_number_propagation(self):
+        """Check if a lot number can be propagated.
+
+        A lot number can be propagated from a component to the finished product if:
+        - the type of the BoM is normal (Manufacture this product)
+        - the finished product is tracked by serial number
+        - the quantity of the finished product is 1 and its UoM is unit
+        - there is at least one bom line, with a component tracked by serial,
+          having a quantity of 1 and its UoM is unit
+        """
+        uom_unit = self.env.ref("uom.product_uom_unit")
+        for bom in self:
+            bom.display_lot_number_propagation = (
+                bom.type == "normal"
+                and bom.product_tmpl_id.tracking == "serial"
```

Comment:
we narrow here only for "serial" tracking would be good to rid of the "lot" notion in naming as implementation only about the propagation of a serial number

Reply:
I can't say about that, maybe the support for lot numbers could be added later? cc @grindtildeath

Reply:
That's the idea. cf ROADMAP.rst